src/vmspawn/vmspawn-util.c

```diff
@@ -0,0 +1,186 @@
+                [ARCHITECTURE_ALPHA]       = "alpha",
+                [ARCHITECTURE_X86_64]      = "x86_64",      /* differs from our name */
+                [ARCHITECTURE_X86]         = "i386",        /* differs from our name */
+                [ARCHITECTURE_LOONGARCH64] = "loongarch64",
+                [ARCHITECTURE_MIPS64_LE]   = "mips",        /* differs from our name */
+                [ARCHITECTURE_MIPS_LE]     = "mips",        /* differs from our name */
+                [ARCHITECTURE_PARISC]      = "hppa",        /* differs from our name */
+                [ARCHITECTURE_PPC64_LE]    = "ppc",         /* differs from our name */
+                [ARCHITECTURE_PPC64]       = "ppc",         /* differs from our name */
+                [ARCHITECTURE_PPC]         = "ppc",
+                [ARCHITECTURE_RISCV32]     = "riscv32",
+                [ARCHITECTURE_RISCV64]     = "riscv64",
+                [ARCHITECTURE_S390X]       = "s390x",
+        };
+
+        FOREACH_STRING(s, "qemu", "qemu-kvm") {
+                r = find_executable(s, ret_qemu_binary);
+                if (r == 0)
+                        return 0;
+
+                if (r != -ENOENT)
+                        return r;
+        }
+
+        const char *arch_qemu = architecture_to_qemu_table[native_architecture()];
+        if (!arch_qemu) {
+                return log_debug_errno(SYNTHETIC_ERRNO(ESRCH), "Architecture %s not supported by qemu", architecture_to_string(native_architecture()));
+        }
+
+        _cleanup_free_ char *qemu_arch_specific = NULL;
+        qemu_arch_specific = strjoin("qemu-system-", arch_qemu);
+        if (!qemu_arch_specific)
+                return -ENOMEM;
+
+        r = find_executable(qemu_arch_specific, ret_qemu_binary);
+        if (r == 0)
+                return 0;
+
+        return -ENOENT;
+}
+
+int find_ovmf_vars(const char **ret_ovmf_vars) {
+#ifdef __x86_64__
+        #define OVMF_VARS_LOCATIONS "/usr/share/ovmf/x64/OVMF_VARS.fd"
+#elif defined(__i386__)
+        #define OVMF_VARS_LOCATIONS "/usr/share/edk2/ovmf-ia32/OVMF_VARS.fd", "/usr/share/OVMF/OVMF32_VARS_4M.fd"
+#elif defined(__arm__)
+        #define OVMF_VARS_LOCATIONS "/usr/share/AAVMF/AAVMF32_VARS.fd"
+#elif defined(__aarch64__)
+        #define OVMF_VARS_LOCATIONS "/usr/share/AAVMF/AAVMF_VARS.fd"
+#endif
+
+#define GENERIC_OVMF_VARS_LOCATIONS \
+        "/usr/share/edk2/ovmf/OVMF_VARS.fd", \
+        "/usr/share/edk2-ovmf/OVMF_VARS.fd", \
+        "/usr/share/qemu/OVMF_VARS.fd", \
+        "/usr/share/ovmf/OVMF_VARS.fd", \
+        "/usr/share/OVMF/OVMF_VARS.fd"
+
+        FOREACH_STRING(location, OVMF_VARS_LOCATIONS , GENERIC_OVMF_VARS_LOCATIONS) {
+                if (access(location, F_OK) == 0) {
+                        if (ret_ovmf_vars)
+                                *ret_ovmf_vars = location;
+                        return 0;
+                }
+        }
+
+        log_error("Couldn't find OVMF UEFI variables file.");
```

Comment:
should be downgraded to debug i guess, and also merge with the return below:

```c
return log_debug_errno(SYNTHETIC_ERRNO(ENOENT), "Couldn't...");
```

Reply:
Fixed in: 5659535f2e